CMakeLists.txt

```diff
@@ -228,7 +227,7 @@ endif()
 message(STATUS "BLAS library found: ${BLAS_LIBRARIES}")
 
 if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "AMD64")
-	if(APPLE OR UNIX)
+    if(NOT CMAKE_OSX_ARCHITECTURES STREQUAL "arm64" AND (APPLE OR UNIX))
```

Comment:
I think the problem is that `CMAKE_OSX_ARCHITECTURES` is not really set and that we are using  `CMAKE_HOST_SYSTEM_PROCESSOR` instead of `CMAKE_SYSTEM_PROCESSOR`, so it incorrectly assumes is in `x86_64`.
I would try with:
```
if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
    if(APPLE OR UNIX)
 ```